### PR TITLE
[bitnami/harbor]: redis sentinel support in harbor

### PR DESCRIPTION
**Pull request by artbegolli**
**Description of the change**

This change enhances the chart to support redis sentinel in line with an enhancement made to harbor components last year https://github.com/goharbor/harbor/pull/11155

**Benefits**

Users have an optional to use redis sentinel as an external redis instance. This is particularly valuable for those looking to deploy Harbor in HA.

**Possible drawbacks**

None

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/harbor/values.yaml [hunk 1]
```diff
@@ -2660,13 +2660,26 @@ redis:
...
+
+  ## Redis(TM) sentinel configuration
+  ##
+  ## If sentinel is enabled the below sentinel configurations are used as the hostname
+  ##
+  sentinelEnabled: false
+  sentinelMasterSet: 'mymaster'
+  ## Configure redis sentinel hostnames in the following pattern
+  ## <host_sentinal1>:<port_sentinel1>,<host_sentinal2>:<port_sentinel2>,<host_sentinal2>:<port_sentinel3>
+  sentinelHosts: ''
```
Comment (carrodher) [inline bitnami/harbor/values.yaml:2681]:
```suggestion
  ## Redis(TM) sentinel configuration
  ## If sentinel is enabled the below sentinel configurations are used as the hostname
  ##
  sentinel:
    enabled: false
    masterSet: 'mymaster'
    ## Configure redis sentinel hostnames in the following pattern
    ## <host_sentinal1>:<port_sentinel1>,<host_sentinal2>:<port_sentinel2>,<host_sentinal2>:<port_sentinel3>
    ##
    hosts: ''
```

File: bitnami/harbor/values.yaml [hunk 2]
```diff
@@ -2660,13 +2660,26 @@ redis:
 ## If the Redis(TM) included in the chart is disabled, Harbor will use below
 ## Redis(TM) parameters to connect to an external Redis(TM) server.
 ##
+## Support for both Redis and Redis+Sentinel
+##
```
Comment (carrodher) [inline bitnami/harbor/values.yaml:2664]:
```suggestion
## If the Redis(TM) included in the chart is disabled, Harbor will use below
## Redis(TM) parameters to connect to an external Redis(TM) server.
## Support for both Redis(TM) and Redis+Sentinel(TM)
##
```

File: bitnami/harbor/README.md [hunk 3]
```diff
@@ -734,6 +734,9 @@ The following tables list the configurable parameters of the Harbor chart and th
 | `redis.slave.persistence.enabled`         | Enable persistence for slave Redis<sup>TM</sup>                                                           | `true`      |
 | `externalRedis.host`                      | Host of the external redis                                                                                | `localhost` |
 | `externalRedis.port`                      | Port of the external redis                                                                                | `6379`      |
+| `externalRedis.sentinelEnabled`           | If external redis with sentinal is used, set it to `true`                                                 | `false`     |
+| `externalRedis.masterSet`                 | Name of sentinel masterSet if sentinel is used                                                            | `mymaster`  |
+| `externalRedis.sentinelHosts`             | Sentinel hosts and ports in the format <host_sentinal1>:<port_sentinel1>,<host_sentinal2>:<port_sentinel2>| `nil`       |
```
Comment (carrodher) [inline bitnami/harbor/README.md:739]:
```suggestion
| `externalRedis.sentinel.enabled`          | If external redis with sentinal is used, set it to `true`                                                 | `false`     |
| `externalRedis.sentinel.masterSet`        | Name of sentinel masterSet if sentinel is used                                                            | `mymaster`  |
| `externalRedis.sentinel.hosts`            | Sentinel hosts and ports in the format <host_sentinal1>:<port_sentinel1>,<host_sentinal2>:<port_sentinel2>| `nil`       |
```

**Review by carrodher — COMMENTED** [review]
Thanks for your contribution, I just added some suggestions to place the sentinel parameters under a `sentinel:` block, can you adapt the _bitnami/harbor/templates/_helpers.tpl_ to reflect those changes?

In the same way, in the _bitnami/harbor/templates/_helpers.tpl_ there are some places where the order of the logic was modified, I mean, currently we've something like
```
  {{- if (include "harbor.redis.escapedRawPassword" . ) -}}
    {{- printf "redis://redis:%s@%s:%s/%s" (include "harbor.redis.escapedRawPassword" . ) (include "harbor.redis.host" . ) (include "harbor.redis.port" . ) (include "harbor.redis.jobserviceDatabaseIndex" . ) -}}
  {{- else -}}
    {{- template "harbor.redis.host" . -}}:{{ template "harbor.redis.port" . -}}/{{ template "harbor.redis.jobserviceDatabaseIndex" . -}}
  {{- end -}}
```
but with the new changes, this part of the code is placed inside an `else` statement, what do you think about maintaining this code as the first condition (top level `if` statement) and then place the new code in the `else`?

File: bitnami/harbor/README.md (hunk 3 above)
Comment (artbegolli) [reply bitnami/harbor/README.md:739]: 👍

**Review by carrodher — APPROVED** [review]
Thanks, the changes LGTM. I just launched some internal tests, if everything works as expected the PR is going to be automatically merged.
The issue in the linter is because the airflow README was updated but the chart version was not bumped, for changes in the README is not a requirement so it's fine.

**Pull request merged**